gui/default/syncthing/device/shareDeviceIdDialogView.html

```diff
@@ -12,21 +12,21 @@
       </span>
       <span translate>You can also copy and paste the content into the message manually.</span>
     </p>
-    <hr>
     <div ng-switch-when="email">
+      <hr>
       <h5 translate>Subject:</h5>
-      <pre style="word-break: normal; white-space: pre-wrap;">{{shareDeviceIdParams.subject}}<button
-        type="button" class="btn btn-default pull-right" ng-click="copyToClipboard(shareDeviceIdParams.subject)" tooltip data-original-title="{{ 'Copy' | translate }}">
-          <span class="fa fa-clone"></span>
-        </button
-      ></pre>
+      <pre style="word-break: normal; white-space: pre-wrap;">{{shareDeviceIdParams.subject}}<button type="button" class="btn btn-default pull-right" ng-click="copyToClipboard(shareDeviceIdParams.subject)" tooltip data-original-title="{{ 'Copy' | translate }}">
+        <span class="fa fa-clone"></span>
+      </button></pre>
       <h5 translate>Body:</h5>
     </div>
-    <pre style="word-break: normal; white-space: pre-wrap;">{{shareDeviceIdParams.body}}<button
-      type="button" class="btn btn-default pull-right" ng-click="copyToClipboard(shareDeviceIdParams.body)" tooltip data-original-title="{{ 'Copy' | translate }}">
-        <span class="fa fa-clone"></span>
-      </button
-    ></pre>
+    <pre style="word-break: normal; white-space: pre-wrap;">{{shareDeviceIdParams.body}}<button type="button" class="btn btn-default pull-right" ng-click="copyToClipboard(shareDeviceIdParams.body)" tooltip data-original-title="{{ 'Copy' | translate }}">
+      <span class="fa fa-clone"></span>
+    </button></pre>
+    <p>In addition, you can download and attach the ID encoded as QR code as well!</p>
+    <div>
+      <a href="qr/?text={{currentDevice.deviceID}}" target="_blank" download="{{currentDevice.deviceID}}.png"><img class="img-thumbnail" ng-src="qr/?text={{currentDevice.deviceID}}" height="109" width="109" alt="{{'QR code' | translate}}"></a>
+    </div>
```

Comment:
Hm... I don't really see the purpose of this.  When sending via SMS, it's going to cause a costly (at least with German providers) MMS instead.  For email, I doubt many people would succeed in scanning a QR code from an email attachment (I wouldn't know how easily).

If there is a legitimate use case to send the QR code as image, it can already be downloaded from the `idqrModalView`.  Whoever needs that will also know how to attach the downloaded file to a message somewhere.  You could make it easier by linking to the downloadable file from the large QR code image, which would save a right click (long press) and a click on "Save image" or similar.  Not sure it's worth it though.

Reply:
This is highly country-dependent, but there are places, where MMS costs the same or almost the same as normal SMS messages. However, you're right that scanning an image located on the device itself would be problematic. Ideally, the QR scanner that's built into the Synching app should have this functionality, but right now it's only capable of using the camera to take a picture of the QR.

The "downloadable" button next to the large QR image isn't a bad idea, I think. This is because non-techy people actually don't know that they can right click and save images.

Reply:
What I meant was to make the image itself clickable (a link), not another separate button.  Just like how your thumbnail works, just on the original.

Reply:
Done.